plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/util/PathComparator.groovy

```diff
@@ -0,0 +1,29 @@
+package com.here.gradle.plugins.jobdsl.util
+
+/**
+ * A comparator that orders path names. Files in the same directory are ordered alphabetically, and always come before
+ * files in subdirectories. Subdirectories are also ordered alphabetically.
+ */
+class PathComparator implements Comparator<String> {
```

Comment:
How about [this code](http://www.java2s.com/Tutorials/Java/Data_Structure_How_to/Sort/Sort_file_path.htm) instead which is a bit shorter?

Reply:
The code is not sorting folders alphabetically, you can see in the example output that `/games/snake` comes after `/usr/local`, with more files and more folders this would become a total mess.